### PR TITLE
feat(discord): Implement Discord slash commands handler

### LINKED ISSUE #50883: Implement Discord slash commands handler in API client
Provide a method in the Discord API client that takes a list of slash commands and ensures the slash commands registered with Discord matches. 

We'll need to 
- ~fetch the currently registered commands~ (not necessary)
- ~delete any commands that are missing from the input list~ (not necessary)
- do a [bulk global command overwrite](https://discord.com/developers/docs/interactions/application-commands#bulk-overwrite-global-application-commands) with the input list

Finally, once this is implemented, make sure we call it in the `setup` method of our Discord `IntegrationProvider`. This way the list of registered commands always matches what we have defined in our repo.

### PR DESCRIPTION
**Pull request by spalmurray**
Programmatically handle command definition and updates. We'll limit command updates to once per hour since they should not be changing frequently at all.

### CODE REVIEW

File: src/sentry/integrations/discord/commands.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        cache_key = "discord-bot-commands-updated"
+        result = cache.get(cache_key)
+
+        if result is None:
+            try:
+                DiscordClient().overwrite_application_commands(self.COMMANDS)
+            except ApiError as e:
+                logger.error("discord.setup.update_bot_commands_failure", extra={"status": e.code})
+
+            cache.set(cache_key, True, 3600)
```
Comment (spalmurray) [inline src/sentry/integrations/discord/commands.py:37]: I think waiting the hour before trying again upon failure is fine. Let me know what you think though!

Comment (snigdhas) [reply src/sentry/integrations/discord/commands.py:37]: This should be alright given how infrequently the list of commands will change. If it does end up failing a substantial amount, we can retry here instead.

File: src/sentry/integrations/discord/requests/base.py [hunk 2]
```diff
@@ -110,3 +110,6 @@ def _error(self, key: str) -> None:
 
     def is_ping(self) -> bool:
         return self._data.get("type", 0) == 1
+
+    def is_command(self) -> bool:
+        return self._data.get("type", 0) == 2
```
Comment (snigdhas) [inline src/sentry/integrations/discord/requests/base.py:115]: If we support more than the ping/command data types, we may want to define a separate enum with the data types and replace this with something like 
```suggestion
        return self._data.get("type", 0) == DISCORD_DATA_TYPE.COMMAND
```

Comment (snigdhas) [reply src/sentry/integrations/discord/requests/base.py:115]: Don't need to make that change in this PR though

**Review by snigdhas — APPROVED** [review]

File: src/sentry/integrations/discord/requests/base.py (hunk 2 above)
Comment (spalmurray) [reply src/sentry/integrations/discord/requests/base.py:115]: Great point. I'll make a note of that!

**Pull request merged**